src/axom/core/Array.hpp

```diff
@@ -196,27 +193,27 @@ class Array : public ArrayBase<T, DIM, Array<T, DIM, SPACE>>
    * \brief Constructor for transferring between memory spaces
    * 
    * \param [in] other The array in a different memory space to copy from
-   * \param [in] allocator_id the ID of the allocator to use (optional)
+   * \param [in] allocator_id the ID of the allocator to use (optional).
+   *     Default value of -1 causes the new Array to use the allocator
```

Comment:
This comment is confusing to me.

* default to use allocator of the other array
* otherwise you specify an allocator and it has to match the other array (then why are you specifying)
* OR you specify and it uses the default allocator, which is the allocator of other or is that the dynamic allocator

I feel like I'm missing something.

Reply:
So the copy constructor behavior is:
- if allocator ID is explicitly-specified, use that, otherwise default to use allocator ID of other array
- as a fallback, use default allocator for the memory space parameter if the allocator ID isn't compatible with the memory space
  - i.e. `Array<T, 1, Host> host_arr(device_arr)` - since `device_arr.getAllocatorID()` isn't compatible with the Host space, we'll use the default host allocator.
  - Note that we warn if the allocator ID is explicitly specified and isn't compatible with the memory space; if the incompatibility is because of an explicitly-specified memory space, we assume the user wants to copy into that space ala the above example.

I'll try and clarify the comments/documentation some more.